File: Data Structures & Algorithms/three-integer-sum/submission-5.cpp
class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums) {
        sort(nums.begin(),nums.end());
    vector<vector<int>> ans;
    for(int i=0;i<nums.size()-2;)
    {
        int target=-nums[i],l=i+1,r=nums.size()-1,I=i;
        cout<<"target: "<<target<<"\n";
        while(l<r)
        {
            int sum=nums[l]+nums[r];
            if(sum<target)
                l++;
            else if(sum>target)
                r--;
            else
            {
                ans.push_back({nums[i],nums[l],nums[r]});
                int L=l,R=r;
                while(l<r && nums[l]==nums[L])
                    l++;
                while(l<r && nums[r]==nums[R])
                    r--;
            }
        }
        while(nums[I]==nums[i])
            I++;
        i=I;
    }
    return ans;
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-5.js
class Solution {
    twoSum(numbers, target) {
        let n=numbers.length;
        let l=0,r=n-1,sum=numbers[l]+numbers[r];
        while(l<r) {
            sum=numbers[l]+numbers[r];
            if(sum>target) r--;
            else if(sum<target) l++;
            else return [l+1,r+1];
        }
        return [-2,-2];
    }
}
